tests/test_main.py

```diff
@@ -0,0 +1,58 @@
+from fastapi.testclient import TestClient
+from main import app
+
+client = TestClient(app)
+
+def test_missing_parameter_array():
+    response = client.post('/top_n', json={'n': '3'})
+    assert response.status_code == 422
+    assert response.json()['detail'][0]['msg'] == 'field required'
+
+def test_missing_parameter_n():
+    response = client.post('/top_n', json={'arr': [10, 30, 20]})
+    assert response.status_code == 422
+    assert response.json()['detail'][0]['msg'] == 'field required'
+
+def test_invalid_type_array():
+    response = client.post('/top_n', json={'n': '3', 'arr': 'arr'})
+    assert response.status_code == 422
+    assert response.json()['detail'][0]['msg'] == 'value is not a valid list'
+
+def test_invalid_type_n():
+    response = client.post('/top_n', json={'n': 'invalid', 'arr': [10, 30, 20]})
+    assert response.status_code == 422
+    assert response.json()['detail'][0]['msg'] == 'value is not a valid integer'
+
+def test_negative_n():
+    response = client.post('/top_n', json={'n': '-1', 'arr': [10, 30, 20]})
+    assert response.status_code == 400
+    assert response.json()['detail'] == 'N must be a positive integer'
+
+def test_n_equals_0():
+    response = client.post('/top_n', json={'n': '0', 'arr': [10, 30, 20]})
+    assert response.status_code == 400
+    assert response.json()['detail'] == 'N must be a positive integer'
+
+def test_n_greater_than_array_size():
+    response = client.post("/top_n", json={"n": '15', "arr": [10, 30, 20, 21, 11, 22, 33, 44, 15, 33, 10, 30]})
+    assert response.status_code == 400
+    assert response.json()['detail'] == 'length of array should be more than n'
+
+def test_valid_input():
+    response = client.post("/top_n", json={"n": '5', "arr": [10, 30, 20, 21, 11, 22, 33, 44, 15, 33, 10, 30]})
+    assert response.status_code == 200

```

Comment:
I prefer to use built-in variables to label their HTTP status codes.
```suggestion
    assert response.status_code == status.HTTP_201_CREATED

```